src/main/java/com/feedzai/commons/sql/abstraction/engine/impl/CockroachDBEngine.java

```diff
@@ -0,0 +1,384 @@
+
+        if (!pks.isEmpty()) {
+            createTable.add(",");
+
+            final String pkName = md5(format("PK_%s", entity.getName()), properties.getMaxIdentifierSize());
+
+            createTable.add("CONSTRAINT");
+            createTable.add(quotize(pkName));
+            createTable.add("PRIMARY KEY");
+            createTable.add("(" + join(pks, ", ") + ")");
+        }
+        // PK end
+
+        createTable.add(")");
+
+        final String createTableStatement = join(createTable, " ");
+
+        logger.trace(createTableStatement);
+
+        Statement s = null;
+        try {
+            s = conn.createStatement();
+            s.executeUpdate(createTableStatement);
+        } catch (final SQLException ex) {
+            if (ex.getSQLState().equals(NAME_ALREADY_EXISTS)) {
+                logger.debug(dev, "'{}' is already defined", entity.getName());
+                handleOperation(new OperationFault(entity.getName(), OperationFault.Type.TABLE_ALREADY_EXISTS), ex);
+            } else {
+                throw new DatabaseEngineException("Something went wrong handling statement", ex);
+            }
+        } finally {
+            try {
+                if (s != null) {
+                    s.close();
+                }
+            } catch (final Exception e) {
+                logger.trace("Error closing statement.", e);
+            }
+        }
+    }
+
+    @Override
+    protected void addPrimaryKey(final DbEntity entity) throws DatabaseEngineException {
+        // Do nothing because Primary Keys are added at creation table time.
+    }
+
+    @Override
+    public Class<? extends AbstractTranslator> getTranslatorClass() {
+        return CockroachDBTranslator.class;
+    }
+
+    @Override
+    public synchronized Long persist(final String name,
+                                     final EntityEntry entry,
+                                     final boolean useAutoInc) throws DatabaseEngineException {
+        ResultSet generatedKeys = null;
+        try {
+            getConnection();
+
+            final MappedEntity me = entities.get(name);
+
+            if (me == null) {
+                throw new DatabaseEngineException(String.format("Unknown entity '%s'", name));
+            }
+
+            final PreparedStatement ps;
+            if (useAutoInc) {
+                ps = me.getInsertReturning();
+            } else {
+                ps = me.getInsertWithAutoInc();
+            }
+
+            entityToPreparedStatement(me.getEntity(), ps, entry, useAutoInc);
+            generatedKeys = ps.executeQuery();
+
+
+            long ret = 0;
+
+            if (useAutoInc) {
+                if (generatedKeys.next()) {
+                    ret = generatedKeys.getLong(1);
+                }
+            } else if (hasIdentityColumn(me.getEntity())) {
+                final List<Map<String, ResultColumn>> q = query(
+                        "SELECT max(" + quotize(me.getAutoIncColumn()) + ") FROM" + quotize(name)
```

Comment:
Isn't this missing a space after FROM? Is this being tested? Using format seems a better choice here.

Reply:
Nice catch!
For now, this is only a cosmetic issue, it works because what follows is always quotized and the quotes also act as a separator (but the space should be there anyway).